packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts

```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
   expect(out.resolvers['Member.team.req.vtl']).toMatchSnapshot();
 });
 
+test('fails if indexName is provided with references', () => {
+  const inputSchema = `
+    type Team @model {
+      id: ID!
+      name: String!
+      members: [Member] @hasMany(indexName: "foo", references: ["teamID"])
+    }
+    type Member @model {
+      id: ID!
+      teamID: ID! @index(name: "foo")
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new IndexTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError('Invalid @hasMany directive on Team.members - indexName is not supported with DDB references.');
```

Comment:
I've got one naive question about how do we determine the GSI to use if no `indexName` is supported?

Reply:
We create it ourselves with the format of `gsi-[TypeName].[FieldName]`. This aligns with what we are doing today when an index name isn't provided for `hasMany` / `belongsTo`.